packages/firestore/src/util/sorted_map.ts

```diff
@@ -440,7 +444,7 @@ export class LLRBNode<K, V> {
 
   // Returns new tree, with the key/value added.
   insert(key: K, value: V, comparator: Comparator<K>): LLRBNode<K, V> {
-    let n = this;
+    let n = this as LLRBNode<K, V>;
```

Comment:
This tripped me up (I didn't realize that 'this' was a special type)...  I wonder if 
```
let n: LLRBNode<K, V> = this;
```
would be less confusing? Eh, don't care strongly.

Reply:
Oh, yeah, I do think that's less confusing. Thanks!